src/tree/AzureDBAPIStep.ts

```diff
@@ -0,0 +1,70 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { AzureWizardExecuteStep, AzureWizardPromptStep, IAzureQuickPickItem, IWizardOptions } from 'vscode-azureextensionui';
+import { API, Experience, getExperienceQuickPicks } from '../AzureDBExperiences';
+import { ext } from '../extensionVariables';
+import { IPostgresWizardContext } from '../postgres/commands/PostgresAccountWizard/IPostgresWizardContext';
+import { PostgresServerConfirmPWStep } from '../postgres/commands/PostgresAccountWizard/PostgresServerConfirmPWStep';
+import { PostgresServerCreateStep } from '../postgres/commands/PostgresAccountWizard/PostgresServerCreateStep';
+import { PostgresServerCredPWStep } from '../postgres/commands/PostgresAccountWizard/PostgresServerCredPWStep';
+import { PostgresServerCredUserStep } from '../postgres/commands/PostgresAccountWizard/PostgresServerCredUserStep';
+import { PostgresServerFirewallStep } from '../postgres/commands/PostgresAccountWizard/PostgresServerFirewallStep';
+import { PostgresServerNameStep } from '../postgres/commands/PostgresAccountWizard/PostgresServerNameStep';
+import { PostgresServerSetCredentialsStep } from '../postgres/commands/PostgresAccountWizard/PostgresServerSetCredentialsStep';
+import { PostgresServerSetFirewallStep } from '../postgres/commands/PostgresAccountWizard/PostgresServerSetFirewallStep';
+import { localize } from '../utils/localize';
+import { CosmosDBAccountCreateStep } from './CosmosDBAccountWizard/CosmosDBAccountCreateStep';
+import { CosmosDBAccountNameStep } from './CosmosDBAccountWizard/CosmosDBAccountNameStep';
+import { ICosmosDBWizardContext } from './CosmosDBAccountWizard/ICosmosDBWizardContext';
+import { IAzureDBWizardContext } from './IAzureDBWizardContext';
+
+export class AzureDBAPIStep extends AzureWizardPromptStep<IPostgresWizardContext | ICosmosDBWizardContext> {
+    public async prompt(wizardContext: IAzureDBWizardContext): Promise<void> {
+        const picks: IAzureQuickPickItem<Experience>[] = getExperienceQuickPicks();
+
+        const result: IAzureQuickPickItem<Experience> = await ext.ui.showQuickPick(picks, {
+            placeHolder: localize('selectDBServerMsg', 'Select an Azure Database Server.')
+        });
+
+        wizardContext.defaultExperience = result.data;
+    }
+
+    public async getSubWizard(wizardContext: IAzureDBWizardContext): Promise<IWizardOptions<IPostgresWizardContext | ICosmosDBWizardContext>> {
+        let promptSteps: AzureWizardPromptStep<IPostgresWizardContext | ICosmosDBWizardContext>[];
+        let executeSteps: AzureWizardExecuteStep<IPostgresWizardContext | ICosmosDBWizardContext>[];
```

Comment:
The way you defined these variables is a little odd to me. In general variables should be defined in the minimal scope that they're used. For example:

Option 1: They're defined  and used _outside_ the if/else cases
```
let promptSteps: AzureWizardPromptStep<IPostgresWizardContext | ICosmosDBWizardContext>[];
let executeSteps: AzureWizardExecuteStep<IPostgresWizardContext | ICosmosDBWizardContext>[];
if () {
    promptSteps = [];
    executeSteps = [];
} else {
    promptSteps = [];
    executeSteps = [];
}
return { promptSteps, executeSteps };
```

Option 2: They're defined  and used _inside_ the if/else cases
```
if () {
    const promptSteps: AzureWizardPromptStep<IPostgresWizardContext>[] = [];
    const executeSteps: AzureWizardExecuteStep<IPostgresWizardContext>[] = [];
    return { promptSteps, executeSteps };
} else {
    const promptSteps: AzureWizardPromptStep<ICosmosDBWizardContext>[] = [];
    const executeSteps: AzureWizardExecuteStep<ICosmosDBWizardContext>[] = [];
    return { promptSteps, executeSteps };
}
```

Reply:
Agreed. This is carried from previous commits, missed to resolve it here.